### PR TITLE
Remove FAR from 32/64bit arch

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary

since these arch doesn't distinguish between near and far pointers

## Impact

minor code refactor only

## Testing

ci

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



No, this PR does not fully meet the NuttX requirements.  While it provides *some* information, it lacks crucial details.

Here's a breakdown of what's missing:

* **Summary:**  The summary is too vague.  It states "since these arch doesn't distinguish between near and far pointers," but doesn't explain *why* the change is necessary. What problem does this solve? What code is being refactored and why?  What is the intended outcome of the refactoring?
* **Impact:** While it claims a "minor code refactor," it doesn't address *any* of the specific impact questions. Even if the changes are minor, the PR author needs to explicitly state "NO" or provide a brief explanation for each impact category (user, build, hardware, documentation, security, compatibility).
* **Testing:** "ci" is insufficient. While Continuous Integration testing is important, the PR needs to demonstrate *specific* local testing results.  This includes:
    * Listing the specific build host and target environments used for testing.  "ci" doesn't tell us which architectures or boards were tested.
    * Providing *actual* test logs (even if they are short) before and after the change to demonstrate the impact of the change.  Simply stating "ci" doesn't offer any evidence of successful testing.


The PR author needs to significantly expand on these sections to meet the NuttX requirements.  Without more details, reviewers cannot effectively evaluate the changes.

File: include/nuttx/can/sja1000.h
Comment (hartmannathan) [inline include/nuttx/can/sja1000.h:1]: This file did not need to change. Was committed by mistake?

File: include/nuttx/can/sja1000.h [hunk 1]
```diff
@@ -155,7 +155,7 @@ struct sja1000_dev_s
  ****************************************************************************/
 
 static inline uint32_t sja1000_getreg(struct sja1000_dev_s *sja_priv,
```
Comment (anchao) [inline include/nuttx/can/sja1000.h:157]:
```suggestion
static inline uint32_t sja1000_getreg(FAR struct sja1000_dev_s *sja_priv,
```

File: include/nuttx/can/sja1000.h [hunk 2]
```diff
@@ -177,7 +177,7 @@ static inline uint32_t sja1000_getreg(struct sja1000_dev_s *sja_priv,
  ****************************************************************************/
 
 static inline void sja1000_putreg(struct sja1000_dev_s *sja_priv,
```
Comment (anchao) [inline include/nuttx/can/sja1000.h:179]:
```suggestion
static inline void sja1000_putreg(FAR struct sja1000_dev_s *sja_priv,
```

File: include/nuttx/can/sja1000.h (hunk 1 above)
Comment (xiaoxiang781216) [reply include/nuttx/can/sja1000.h]: drop this change since it's added by accident.

**Review by anchao — APPROVED** [review]

**Review by jasonbu — APPROVED** [review]

**Pull request merged**